Add documentation for environment exporter plugin hooks

jezdez opened the pull request:
This documentation covers the new environment exporter plugin system that was implemented in the conda codebase. It includes:

- New environment_exporters plugin documentation
- Updated export command documentation
- API reference updates
- User guide updates for environment management
- Glossary and cheatsheet updates

The feature implementation was merged separately, this commit only adds the corresponding documentation.

See PR docs build at https://continuumio-conda-docs--15051.com.readthedocs.build/projects/conda/en/15051/.

<!-- Hello! Thanks for submitting a PR! To help make things go a bit more
     smoothly, we would appreciate it if you follow this template. -->

### Description

<!-- Good things to put here include:
       - reasons for the change (please link any relevant issues!),
       - any noteworthy (or hacky) choices to be aware of,
       - or what the problem resolved here looked like. -->

### Checklist - did you ...

<!-- If any of the following items aren't relevant to your contribution,
     please either tick them or use ~strikethrough~ so we know you've gone
     through the checklist. -->

- [x] Add a file to the `news` directory ([using the template](https://github.com/conda/conda/blob/main/news/TEMPLATE)) for the next release's release notes?
     <!-- All "significant" changes should get an entry:
            - user-facing changes or enhancements
            - bug fixes
            - deprecations
            - documentation updates
            - etc -->
- [ ] Add / update necessary tests?
- [x] Add / update outdated documentation?


<!-- Just as a reminder, everyone in all conda org spaces (including PRs)
     must follow the Conda Org Code of Conduct (link below).

     Finally, once again, thanks for your time and effort. If you have any
     feedback in regards to your experience contributing here, please
     let us know!

     Helpful links:
       - Conda Org COC: https://github.com/conda/conda/blob/main/CODE_OF_CONDUCT.md
       - Contributing docs: https://github.com/conda/conda/blob/main/CONTRIBUTING.md -->

jezdez committed:
Add documentation for environment exporter plugin hooks

This documentation covers the new environment exporter plugin system
that was implemented in the conda codebase. It includes:

- New environment_exporters plugin documentation
- Updated export command documentation
- API reference updates
- User guide updates for environment management
- Glossary and cheatsheet updates

The feature implementation was merged separately, this commit only
adds the corresponding documentation.

jezdez committed:
rST typo.

jezdez committed:
rST fix.

jezdez committed:
rST fix.

kenodegard commented on docs/source/commands/env/export.rst:
```diff
@@ -1,10 +1,65 @@
+The ``conda env export`` command exports conda environments to YAML format.
+This command continues to be fully supported.
+
+For new projects, you may want to consider :doc:`conda export <../export>` which provides additional features:
```
```suggestion
For new projects, consider :doc:`conda export <../export>` which provides additional features:
```

kathatherine commented on docs/source/dev-guide/plugins/environment_exporters.rst:
```diff
@@ -0,0 +1,411 @@
+
+.. code-block:: text
+
+    my-conda-export-plugin/
+    ├── pyproject.toml
+    ├── my_export_plugin/
```
Layperson question, but what's the difference between the `my-conda-export-plugin` and `my_export_plugin` directories? Obviously one is a parent to the other, but is there a naming convention that's being followed here that I am not aware of? Does it matter that conda is missing from `my_export_plugin`?

kathatherine commented on docs/source/glossary.rst:
```diff
@@ -40,6 +40,81 @@ environment with just ``program_name``.
+
+A CEP 23 compliant environment file format that contains MatchSpec
+strings for conda-compatible package specifications. This format is
+similar to pip's requirements.txt but uses conda's MatchSpec syntax.
```
```suggestion
similar to pip's ``requirements.txt`` but uses conda's MatchSpec syntax.
```

kathatherine reviewed (commented):
A few suggestions and a question from me

jezdez committed:
Apply suggestions from code review

Co-authored-by: Katherine Kinnaman <kkinnaman@anaconda.com>
Co-authored-by: Ken Odegard <kodegard@anaconda.com>
Co-authored-by: Travis Hathaway <travis.j.hathaway@gmail.com>

jezdez committed:
Apply suggestions from code review

jezdez committed:
Updated the documentation for various environment export formats (YAML, JSON, explicit, and requirements) to improve clarity and consistency. The format details are now presented in a structured table format, enhancing readability and user understanding of the available options and their aliases.

jezdez committed:
Sort list.

jezdez committed:
Add sample output to json exporter.

jezdez committed:
Use real requirements output.

jezdez committed:
Real example for explicit format.

jezdez committed:
Fix YAML and JSON output.

jezdez committed:
Add warning about differences in requirements format

Clarified that the requirements format produces conda MatchSpec strings, which include full package specifications, unlike the simpler output from `conda list --export`. This addition enhances user understanding of the export formats.

jezdez committed:
Remove the Plugin Registration and Distribution section since it overlaps with the quickstart, which I can link to instead.

jezdez committed:
Add news.

kenodegard commented on docs/source/commands/export.rst:
```diff
@@ -0,0 +1,337 @@
+   # platform: osx-64
+   @EXPLICIT
+   https://repo.anaconda.com/pkgs/main/osx-64/python-3.9.7-h88f2d9e_0.tar.bz2
+   https://repo.anaconda.com/pkgs/main/osx-64/numpy-1.21.0-py39h2e5f516_0.tar.bz2
```
I think I was confusing `--explicit` and `--export`
